### PR TITLE
Build Docker image directly from source instead of from PyPI

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

There is a delay (sometime hours) before the newly published package on PyPI becomes available. As a result, the current Docker image build process ends up installing the previous version.

### 🤓 What?

Refactor Dockerfile and the CD pipeline to install directly from the latest source and tag it with the correct version number.

### 🧪 Tested?

CD: https://github.com/MetaphorData/connectors/actions/runs/2599490786
Published: https://hub.docker.com/r/metaphordata/connectors/tags

### CODE REVIEW

**Review by alyiwang — COMMENTED** [review]
Otherwise, LGTM

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -19,9 +19,35 @@ defaults:
...
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.get-version.outputs.version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
```
Comment (alyiwang) [inline .github/workflows/cd.yml:33]: NODE_VERSION: 14 -> 16

Comment (mars-lan) [reply .github/workflows/cd.yml:33]: Updated

**Review by alyiwang — APPROVED** [review]
LGTM

**Pull request merged**